runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,362 @@
+local M = {}
+
+---@private
+local function count_lines(bufnr)
+  return vim.api.nvim_buf_line_count(bufnr)
+end
+
+---@private
+local function get_lines(bufnr, start_lnum, end_lnum)
+  end_lnum = end_lnum or start_lnum
+  local lines = vim.api.nvim_buf_get_lines(bufnr, start_lnum - 1, end_lnum, false)
+  return table.concat(lines) or ""
+end
+
+---@private
+local function iter_lines(bufnr, start_lnum, end_lnum)
+  end_lnum = end_lnum or start_lnum
+  return ipairs(vim.api.nvim_buf_get_lines(bufnr, start_lnum - 1, end_lnum, false))
+end
+
+function M.asm()
+
+end
+
+function M.asm_syntax()
+
+end
+
+function M.bas()
+
+end
+
+function M.bindzone()
+
+end
+
+function M.btm(_, bufnr)
+  if vim.g.dosbatch_syntax_for_btm ~= nil and vim.g.dosbatch_syntax_for_btm ~= 0 then
```

Comment:
```suggestion
  if vim.g.dosbatch_syntax_for_btm ~= 0 then
```
(`0` is truthy in Lua, and `nil ~=0`)

Reply:
But if it is `nil`, shouldn't that set the filetype to `btm`? That won't happen if we omit the first check here.

Reply:
Ah, you're right, I missed that (sorry!)

You can still drop the `~=nil`, though :)